src/display/node_stream.js

```diff
@@ -0,0 +1,384 @@
+  }
+}
+
+class PDFNodeStreamRangeReader extends BaseRangeReader {
+  constructor(stream, start, end) {
+    super(stream);
+
+    this._readableStream = null;
+    let rangeStr = start + '-' + (end - 1);
+    stream.httpHeaders['Range'] = 'bytes=' + rangeStr;
+
+    let options = {
+      host: this._url.host,
+      path: this._url.path,
+      method: 'GET',
+      headers: stream.httpHeaders,
+    };
+    let handleResponse = (response) => {
+      this._readableStream = response;
+
+      response.on('readable', () => {
+        this._readCapability.resolve();
+      });
+
+      response.on('end', () => {
+        response.destroy();
+        this._done = true;
+        this._readCapability.resolve();
+      });
+
+      response.on('error', (reason) => {
+        this._errored = true;
+        this._reason = reason;
+        this._readCapability.resolve();
+      });
+    };
+    this._request = this._url.protocol === 'http:' ?
+      http.request(options, handleResponse) :
+      https.request(options, handleResponse);
+
+    this._request.on('error', (reason) => {
+      this._errored = true;
+      this._reason = reason;
+    });
+    this._request.end();
+  }
+
+  read() {
+    return this._readCapability.promise.then(() => {
+      if (this._done) {
+        return Promise.resolve({ value: undefined, done: true, });
+      }
+      if (this._errored) {
+        return Promise.reject(this._reason);
+      }
+
+      let chunk = this._readableStream.read();
+      if (chunk === null) {
+        this._readCapability = createPromiseCapability();
+        return this.read();
+      }
+      this._loaded += chunk.length;
+      if (this.onProgress) {
+        this.onProgress({ loaded: this._loaded, });
+      }
+      return Promise.resolve({ value: chunk, done: false, });
+    });
+  }
+
+  cancel(reason) {
+    this._readableStream.destroy(reason);
+  }
+}
+
+class PDFNodeStreamFsFullReader extends BaseFullReader {
+  constructor(stream) {
+    super(stream);
+
+    this._readableStream = fs.createReadStream(this._url.path);
+
+    fs.lstat(this._url.path, (error, stat) => {
+      if (error) {
+        this._errored = true;
+        this._reason = error;
+        this._headersCapability.reject(error);
+        return;
+      }
+      this._length = stat.size;
+      this._headersCapability.resolve();
+    });
+
+    this._readableStream.on('readable', () => {
+      this._readCapability.resolve();
+    });
+
+    this._readableStream.on('end', () => {
+      this._readableStream.destroy();
+      this._done = true;
+      this._readCapability.resolve();
+    });
+
+    this._readableStream.on('error', (reason) => {
+      this._errored = true;
+      this._reason = reason;
+      this._readCapability.resolve();
+    });
+  }
+}
+
+class PDFNodeStreamFsRangeReader extends BaseRangeReader {
+  constructor(stream, start, end) {
+    super(stream);
+
+    this._readableStream = fs.createReadStream(this._url.path, { start, end, });
```

Comment:
as mentioned above, use `this._setReadableStream(fs.createReadStream(this._url.path));`